### PR TITLE
Run examples tests on devices

### PR DESCRIPTION
**Pull request by lloydsheng**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'

### Summary of changes

* Enable run examples testes on device farm
* Add examples tests for steve and nightly build

### User impact (optional)

No

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -391,9 +426,13 @@ commands:
...
 
-    access-token-xcconfig:
+    access-token-configs:
         steps:
-            - run: echo "MAPBOX_ACCESS_TOKEN = ${MAPBOX_ACCESS_TOKEN}" > ./developer.xcconfig
+            - run:
+                name: Save mapbox access token to configs
+                command: |
+                    echo "MAPBOX_ACCESS_TOKEN = ${MAPBOX_ACCESS_TOKEN}" > ./developer.xcconfig
+                    echo "${MAPBOX_ACCESS_TOKEN}" > ~/.mapbox
```
Comment (julianrex) [inline .circleci/config.yml:435]: Ideally, we would choose one way or the other. We can address in a follow-up.

File: scripts/device-farm/devicefarm.py [hunk 2]
```diff
@@ -319,7 +319,6 @@ def schedule_run(self, app_package_arn, test_package_arn, test_spec_arn, outputF
 
         aws_command = ['aws', 'devicefarm', 'schedule-run',
                                             '--project-arn', self.project_arn,
-                                            '--configuration', 'customerArtifactPaths={iosPaths=["Documents"]}',
```
Comment (julianrex) [inline scripts/device-farm/devicefarm.py:322]: Why is this removed? Can we re-add? This allows us to pull artifacts from the applications, though admittedly we don't have any yet.

File: scripts/device-farm/testspec.yml [hunk 3]
```diff
@@ -21,32 +21,23 @@ phases:
...
-      - cd /tmp/test-root && xcodebuild test-without-building -destination id=$DEVICEFARM_DEVICE_UDID -xctestrun device.xctestrun -derivedDataPath /tmp/build -resultBundlePath /tmp/build/$SCHEME.$CONFIG.$DATE.xcresult
-
-  # The post test phase includes are commands that are run after your tests are
-  # executed.
-  post_test:
-    commands:
-      # We created a separate /tmp/build directory as we were seeing some
-      # "permission denied" errors with the xcresult package (resulting in an
-      # essentially empty result folder).
-      - cp -R /tmp/build $DEVICEFARM_LOG_DIR
```
Comment (julianrex) [inline scripts/device-farm/testspec.yml:49]: Why is this removed? Do we still get the xcresult files from test runs & example runs?

File: scripts/device-farm/testspec.yml [hunk 4]
```diff
@@ -56,7 +47,3 @@ artifacts:
   # By default, Device Farm will collect your artifacts from following
   # directories
   - $DEVICEFARM_LOG_DIR
-
-  # This appears necessary, even though it is specified on the schedule-run
-  # command-line
-  - Documents
```
Comment (julianrex) [inline scripts/device-farm/testspec.yml:62]: Same question here.

**Review by julianrex — COMMENTED** [review]
This looks great - just a couple of questions about whether we get the data out we expect (mostly the xcresult packages).

File: .circleci/config.yml (hunk 1 above)
Comment (lloydsheng) [reply .circleci/config.yml:435]: We can fix it with a further PR.

File: scripts/device-farm/testspec.yml (hunk 3 above)
Comment (lloydsheng) [reply scripts/device-farm/testspec.yml:49]: I've found the script works well without create `/tmp/build` indeed.

**Review by julianrex — APPROVED** [review]
Looks good - thank you!

File: Makefile [hunk 5]
```diff
@@ -7,6 +7,7 @@ CONFIGURATION    ?= Debug
 BUILD_DIR        ?= $(CURDIR)/build
 JOBS             ?= $(shell sysctl -n hw.ncpu)
 DESTINATIONS     ?= -destination 'platform=iOS Simulator,OS=latest,name=iPhone 11'
+APP_NAME		 ?= $(SCHEME)
```
Comment (neelmistry94) [inline Makefile:10]: nit: align spacing

**Review by neelmistry94 — APPROVED** [review]
One nit comment.
